Add docs for Hive metadata caching config (#21875)
trinodb/trino: docs/src/main/sphinx/object-storage/metastores.md

  - Enable caching for partition metadata. You can disable caching to avoid
    inconsistent behavior that results from it.
  - `true`
* - `hive.metastore-cache.cache-missing`

>   - Enable metadata caching for missing tables.

Reviewer: whats a missing table? I think we need to describe this more.

Reviewer 2: This is about caching absence of objects, i.e., whether we will remember that an object was not found and avoid going back to metstore for it, or always ask metastore for it as long as the object remains not found.

  - `true`
* - `hive.metastore-cache.cache-missing-partitions`

>   - Enable metadata caching for tables that are lacking partition metadata.

Reviewer: are the partitions missing or the metadata is missing?

  - `false`
* - `hive.metastore-cache.cache-missing-stats`

>   - Enable metadata caching for tables that do not have table statistics.

Reviewer: Also confusing..

  - Enable caching for partition metadata. You can disable caching to avoid
    inconsistent behavior that results from it.
  - `true`
* - `hive.metastore-cache.cache-missing`

>   - Enable caching that a table is missing to prevent future metastore calls for

Reviewer: maybe - wdyt @colebow ? its longer but also clearer .. at least in my opinion

```suggestion
  - Enable caching the fact that a table is missing to prevent future metastore calls for
```

    that table.
  - `true`
* - `hive.metastore-cache.cache-missing-partitions`

>   - Enable caching that a partition is missing to prevent future metastore calls

Reviewer: ```suggestion
  - Enable caching the fact that a partition is missing to prevent future metastore calls
```

    for that partition.
  - `false`
* - `hive.metastore-cache.cache-missing-stats`

>   - Enable metadata caching that a table does not have table statistics.

Reviewer: Similar to the others instead of trying this being different: 

```suggestion
  - Enable caching the fact that table statistics for a specific table are missing to prevent future metastore calls.
```
